lib/tenji_maker.rb

```diff
@@ -1,43 +1,38 @@
+# frozen_string_literal: true
+
+TENJI_3BIT_TABLE = {
+  '0' => '---',
+  '1' => '--o',
+  '2' => '-o-',
+  '3' => '-oo',
+  '4' => 'o--',
+  '5' => 'o-o',
+  '6' => 'oo-',
+  '7' => 'ooo'
+}.freeze
+
+# 母音と子音を3bit単位で判別するために、点の有無を「① ④ ② ⑤ ③ ⑥ 」の順に列指向で並べている
 class TenjiMaker
   def to_tenji(text)
-    # 以下はサンプルの仮実装なので、このcase文は全部消して自作ロジックに書き直すこと
-    case text
-    when 'A HI RU'
-      <<~TENJI.chomp
-        o- o- oo
-        -- o- -o
-        -- oo --
-      TENJI
-    when 'KI RI N'
-      <<~TENJI.chomp
-        o- o- --
-        o- oo -o
-        -o -- oo
-      TENJI
-    when 'SI MA U MA'
-      <<~TENJI.chomp
-        o- o- oo o-
-        oo -o -- -o
-        -o oo -- oo
-      TENJI
-    when 'NI WA TO RI'
-      <<~TENJI.chomp
-        o- -- -o o-
-        o- -- oo oo
-        o- o- o- --
-      TENJI
-    when 'HI YO KO'
-      <<~TENJI.chomp
-        o- -o -o
-        o- -o o-
-        oo o- -o
-      TENJI
-    when 'KI TU NE'
-      <<~TENJI.chomp
-        o- oo oo
-        o- -o o-
-        -o o- o-
-      TENJI
+    tenji_serials = text.split(' ').map { |romaji| make_tenji(romaji) }
+    [0..1, 2..3, 4..5].map do |range|
+      tenji_serials.map { |tenji_serial| tenji_serial[range] }.join(' ')
+    end.join("\n")
```

Comment:
rangeを3つ書くよりも、こんな風にした方が少しシンプルでDRYになるかも？
```suggestion
    0.step(4, 2).map do |n|
      tenji_serials.map { |tenji_serial| tenji_serial[n, 2] }.join(' ')
    end.join("\n")
```

Reply:
rangeの3つはDRYでしたね〜。ご提案ありがとうございます😄